### PR TITLE
Conditional operations

### LINKED ISSUE #237: Missing support for conditional S3 operations
Hey, been testing ZeroFS with s3d and looks like we are missing support for conditional s3 operations, zerofs uses [If-None-Match](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html#API_PutObject_RequestSyntax) for split brain protection. While in this software i can use redis when conditional requests are not available it would be nice to have that built in.

Zerofs docs: https://www.zerofs.net/docs/durability#conditional-writes-and-fencing
Conditional requests: https://docs.aws.amazon.com/AmazonS3/latest/userguide/conditional-requests.html

### PR DESCRIPTION
**Pull request by chris124567**
Close #237

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -209,6 +209,19 @@ AWS Signature V4 exclusively. SigV4A is not implemented. Supported
...
+the copy source of `CopyObject` and `UploadPartCopy` honors the matching
+`x-amz-copy-source-if-*` headers.
+
+`PutObject`, `CopyObject` and `CompleteMultipartUpload` honor `If-Match` and
+`If-None-Match` against the current version of the destination, evaluated
+atomically with the write. `DeleteObject` honors `If-Match` (including the `*`
+wildcard), `x-amz-if-match-size` and `x-amz-if-match-last-modified-time`, and
+each object of a multi delete honors its `ETag`, `Size` and `LastModifiedTime`
+elements, where a failed precondition fails only that object. A conditional
+delete with nothing to delete is a no-op, as an unconditional one is.
```
[the remark this answers is not in the document]
Comment (chris124567) [reply README.md:223]: fixed

**Review by peterjan — COMMENTED** [review]
Could you open a PR with extended tox tests for this feature? Often times when we do that one or two changes are necessary before the PR can be merged because of some exotic edge case the tox tests cover.

**Comment by peterjan** [comment]
> Could you open a PR with extended tox tests for this feature? Often times when we do that one or two changes are necessary before the PR can be merged because of some exotic edge case the tox tests cover.

I already have the diff so I might as well copy it here

```
--- a/s3/preconditions.go
+++ b/s3/preconditions.go
@@ -19,8 +19,9 @@ func etagMatches(condition, etag string) bool {
      if condition == wildcardETag {
              return true
      }
+     etag = strings.Trim(etag, `"`)
      for _, v := range strings.Split(condition, ",") {
-             if strings.TrimSpace(v) == etag {
+             if strings.Trim(strings.TrimSpace(v), `"`) == etag {
                      return true
              }
      }

```

with that we can enable conditional tests, see https://github.com/SiaFoundation/s3-tests/pull/11

**Review by peterjan — APPROVED** [review]

**Review by ChrisSchinnerl — APPROVED** [review]
